### ISSUE TITLE
Tests game customisation 

### ISSUE THREAD

**Issue by barbayjuliette** [akolgano/ft_transcendence#86]
Test in branch game-custom or in master once it is merged.
Modal end of game: if you lose against AI, it is a different modal.
Game customizations for simple games, AI, and tournaments.
Game customizations:

Power-ups:
You can slow down the ball for 3 seconds.
Key A and Key O. (let me know if those keys are easy to use for players, cause my keyboard is different)
3 power-ups each. You can see how many are left at the bottom of the game board.
Only one power-up can be activated at the time
If someone scores, the power-up stops.
When testing you can see when the speed of the ball in the console.
AI also uses its power-ups. Check if the frequency of using power-ups is okay (not too fast, etc).
AI uses power-ups only once the ball reaches a certain speed.

Paddle size:
You can choose between 3 sizes of paddle. Default is medium

Practice mode:
The ball doesn't increase speed after every paddle hit. Constant slow speed.

Default game

All the options can be combined, except the default game.

Data is stored in localStorage. So if you refresh during the game, you do not lose the game session, but you lose the score. I will add the score to localStorage, but not the ball position. localStorage cleared only after the game / logout

Try navigating once you have already started a game session. Back and forward buttons. Links.
Check tooltips, with translations.

Test many games, AI, and tournaments with different combinations of game customizations. Everything in both browsers.

Read the description in the subject and let me know what you think of the game customizations and if you think that's okay. At first, I was thinking of doing an attack or option to choose how many points to win, but I think what we have might be enough

I still need to do:

Translations for the game
I will redo the design of the registrations, to have more detailed explanations on how to play.
Up and Down arrows: change to something else

**Comment by chinwenkai** [comment]
![image](https://github.com/user-attachments/assets/22cba688-9d0b-4a1e-851b-ce4dfcb543cd)
there are times the paddle does not show when choosing small. But it is not repeatable. most of the time, it shows.

**Comment by chinwenkai** [comment]
Whenever I select and deselect "default game", and choose another options and play. The game will not have paddle visible

**Comment by chinwenkai** [comment]
[Screencast from 25-10-2024 13:59:08.webm](https://github.com/user-attachments/assets/7b55bc13-a658-4c55-a4ca-9d016416bbb4)

**Comment by barbayjuliette** [comment]
> ![image](https://private-user-images.githubusercontent.com/140361722/380027703-22cba688-9d0b-4a1e-851b-ce4dfcb543cd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk4MzY0MzIsIm5iZiI6MTcyOTgzNjEzMiwicGF0aCI6Ii8xNDAzNjE3MjIvMzgwMDI3NzAzLTIyY2JhNjg4LTlkMGItNGExZS04NTFiLWNlNGRmY2I1NDNjZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyNVQwNjAyMTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05YTFmYjQ3NjI0YTUyMWY4ZjZiNDcxNTQ0MmI5ZmFhN2E4MDVjY2IzYWJkZDcxMWE2YjMwNTM5Mjc0N2U1ODhjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.IWp8cVSnUiy1YXa_5T-1U8igyFTDP_erZ8N-EXmArkk) there are times the paddle does not show when choosing small. But it is not repeatable. most of the time, it shows.

Was this also when you chose default, then deselected default? Or a separate issue?  If yes, only with AI?

**Comment by chinwenkai** [comment]
Whenever I select and deselect default, and choose any other option, IA or power up or practice. It will have the no paddle issue when I press Play

**Comment by barbayjuliette** [comment]
@chinwenkai I just fixed it, let me know if it is okay now; Check if you can find any other case where the paddle is not showing.

**Comment by chinwenkai** [comment]
Whenever I close and open new window to login. The play button will not work on the starting play page. But it will work when I go to another tab such as dashboard or friends, and go back to play.

**Comment by chinwenkai** [comment]
The paddle issue has been resolved :)

**Comment by barbayjuliette** [comment]
> Whenever I close and open new window to login. The play button will not work on the starting play page. But it will work when I go to another tab such as dashboard or friends, and go back to play.

This should be fixed.
Now the default option should grey out other field when navigating the first time / to home and play button should work

**Comment by chinwenkai** [comment]
Ok. it is fixed

**Comment by chinwenkai** [comment]
I think the customisation is sufficient. The power ups, paddle size and practice speed maintains and works ok.
